web/viewer.js

```diff
@@ -1300,6 +1300,35 @@ var PDFViewerApplication = {
 
     var body = document.querySelector('body');
     body.setAttribute('data-mozPrintCallback', true);
+
+    if (!this.hasEqualPageSizes) {
+      console.warn('Not all pages have the same size. The printed result ' +
+          'may be incorrect!');
+    }
+
+    // Insert a @page + size rule to make sure that the page size is correctly
+    // set. Note that we assume that all pages have the same size, because
```

Comment:
Gotcha. I understand that variable pages won't work but what about PDFs with pages that have landscape and portrait dimensions that other than a simple 90deg rotation will have the same dimensions? Anecdotally, I feel that these types of PDFs are common.